numba/tests/test_sysinfo.py

```diff
@@ -0,0 +1,184 @@
+        vm.available = self.mem_available
+        if platform.system() in ('Linux', 'Windows',):
+            # cpu_affiniy only available on Linux and Windows
+            proc = nsi.psutil.Process.return_value
+            proc.cpu_affinity.return_value = self.cpus_list
+        else:
+            nsi.psutil.Process.return_value = None
+
+        self.info = nsi.get_os_spec_info(platform.system())
+
+    def tearDown(self):
+        super(TestSysInfoWithPsutil, self).tearDown()
+        nsi._psutil_import = self.psutil_orig_state
+
+    def test_has_all_data(self):
+        keys = (nsi._mem_total, nsi._mem_available)
+        for k in keys:
+            with self.subTest(k=k):
+                self.assertIn(k, self.info.keys())
+                self.assertIsInstance(self.info[k], int)
+
+    def test_has_correct_values(self):
+        self.assertEqual(self.info[nsi._mem_total], self.mem_total)
+        self.assertEqual(self.info[nsi._mem_available], self.mem_available)
+
+    @skipUnless(platform.system() in ('Linux', 'Windows'),
+                "CPUs allowed info only available on Linux and Windows")
+    def test_cpus_list(self):
+        self.assertEqual(self.info[nsi._cpus_allowed], len(self.cpus_list))
+        self.assertEqual(self.info[nsi._cpus_list],
+                         ' '.join(str(n) for n in self.cpus_list))
+
+
+class TestSysInfoWithoutPsutil(TestCase):
+
+    def setUp(self):
+        super(TestSysInfoWithoutPsutil, self).setUp()
+        self.psutil_orig_state = nsi._psutil_import
+        nsi._psutil_import = False
+        self.info = nsi.get_os_spec_info(platform.system())
+
+    def tearDown(self):
+        super(TestSysInfoWithoutPsutil, self).tearDown()
+        nsi._psutil_import = self.psutil_orig_state
+
+    def test_has_all_data(self):
+        keys = (nsi._mem_total, nsi._mem_available)
+        for k in keys:
+            with self.subTest(k=k):
+                self.assertIn(k, self.info.keys())
+                self.assertIsInstance(self.info[k], int)
+
+
+class TestPlatformSpecificInfo(TestCase):
+
+    def setUp(self):
+        self.plat_spec_info = {
+            'Linux': {
+                str: (nsi._libc_version,),
+            },
+            'Windows': {
+                str: (nsi._os_spec_version,),
+            },
+            'Darwin': {
+                str: (nsi._os_spec_version,),
+            },
+        }
+        self.os_name = platform.system()
+        self.contents = self.plat_spec_info.get(self.os_name, {})
+        self.info = nsi.get_os_spec_info(self.os_name)
+
+    def test_has_all_data(self):
+        keys = chain(*self.contents.values())
+        for k in keys:
+            with self.subTest(k=k):
+                self.assertIn(k, self.info.keys())
+
+    def test_content_type(self):
+        for t, keys in self.contents.items():
+            for k in keys:
+                with self.subTest(k=k):
+                    self.assertIsInstance(self.info[k], t)
```

Comment:
Just spotted that this file is missing:
https://github.com/numba/numba/blob/807689cccaed38272d38df9b0b0cff16c1df59f8/numba/tests/test_alignment.py#L39-L40
at the bottom.

Reply:
Good spot thanks, I added it in e75937f

Reply:
thanks, think you'll need to import `unittest`